### DISCUSSION TITLE
Are you aware of GUT and WAT testing frameworks?

### DISCUSSION THREAD

**Discussion by clemens-tolboom** [MikeSchulze/gdUnit3#9] [FAQ] [answered]
I found 3 tools to do unit tests and more.

I use GUT for a while which has a long track record so am reluctant to change tools.

I wonder what your tool does different/better then one of the others.

Your screenshot with progress bar/integration looks better then the GUT scene to run. GUT has a command-line which I use when committing to run all tests. I will study your **integration** part for sure.

### GUT
- https://github.com/bitwes/Gut
- Stats: 655/41
### WAT
- https://github.com/AlexDarigan/WAT-GDScript
- Stats: 142/4
### gdUnit3
- https://github.com/MikeSchulze/gdUnit3
- Stats: 0/0

**Comment by MikeSchulze** [comment, marked as answer]
Hi clenens,
thanks for your response.

I tested GUT and WAT and i couldn't really get used to it. So i started an half year ago to write my own API. Starting just for fun and get more detailed in the GdScript stuff.

At the end i want to have a modern and easy to use API

Means fully integrated in the editor,
to have **fluentable syntax**
easy navigate to failing tests or create tests
fully mock support to use functions with **default arguments**, **static** function mocking
spy with **argument matcher** for easyier testing
etc,


I current working on CLI tool, with html report. It will be release hopefully the next days.
Here an preview of comming tool.
![image](https://user-images.githubusercontent.com/347037/109431021-a31d5b00-7a04-11eb-838b-83b8d8eef7b4.png)
The image is just a template and the counters is random values for testing ;)

And Stats: 0/0 sure it is new ;)

**Reply by SaschaLucius** [reply]
Hi Mike, 
I would like to see a comparison of the available Unit test libraries. 
Maybe you can add this to the documentation, or I try to prepare something.

**Comment by MikeSchulze** [comment]
You can do this for yourself, but I don't want to compare the APIs which can lead to some kind of bad conversation.
At lease we have the feature-list you can compare.

**Comment by MikeSchulze** [comment]
@clemens-tolboom 
hello, it's been a long time. Are you still interested in the unit test framework GdUnit3?
I am preparing version 2.0.0 with C#(beta) test support.
There will also be a new documentation via GitHub pages and the documentation is now also in the repository.
https://mikeschulze.github.io/gdUnit3/
